src/fides/api/service/storage/streaming/smart_open_streaming_storage.py

```diff
@@ -294,23 +210,75 @@ def _create_attachment_content_stream(
         Returns:
             Iterator that yields chunks of the attachment content
         """
+        for arg in [bucket, key, storage_key]:
+            if arg is None or arg == "":
+                logger.error(f"{arg} cannot be empty: {arg}")
+                raise ValueError(f"{arg} cannot be empty")
+
         try:
             with self.storage_client.stream_read(bucket, key) as content_stream:
                 # Stream in chunks instead of reading entire file
-                chunk_count = 0
-                total_bytes = 0
-                while True:
-                    chunk = content_stream.read(self.chunk_size)
+                chunk_count = total_bytes = 0
+                max_chunks = 100000  # Safety limit to prevent infinite loops
```

Comment:
This should be calculated instead of hard-coded. It's possible to misconfigure this in a way that we can't download a file of size `LARGE_FILE_THRESHOLD` with 100000 chunks of size `self.chunk_size`

`max_chunks = LARGE_FILE_THRESHOLD // self.chunk_size + 1`